airbyte-server/src/main/java/io/airbyte/server/ConfigDumpExport.java

```diff
@@ -0,0 +1,152 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.server;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.io.Archives;
+import io.airbyte.commons.lang.CloseableConsumer;
+import io.airbyte.commons.lang.Exceptions;
+import io.airbyte.commons.yaml.Yamls;
+import io.airbyte.scheduler.persistence.JobPersistence;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.io.FileUtils;
+
+// TODO: Write a test case which compares the output dump with the output of ArchiveHandler export
+// for the same data
+public class ConfigDumpExport {
+
+  private static final String ARCHIVE_FILE_NAME = "airbyte_config_dump";
+  private static final String CONFIG_FOLDER_NAME = "airbyte_config";
+  private static final String DB_FOLDER_NAME = "airbyte_db";
+  private static final String VERSION_FILE_NAME = "VERSION";
+
+  private final ConfigDumpUtil configDumpUtil;
+  private final JobPersistence jobPersistence;
+  private final String version;
+
+  public ConfigDumpExport(Path storageRoot, JobPersistence jobPersistence, String version) {
+    this.configDumpUtil = new ConfigDumpUtil(storageRoot);
+    this.jobPersistence = jobPersistence;
+    this.version = version;
+  }
+
+  public File dump() {
+    try {
+      final Path tempFolder = Files.createTempDirectory(Path.of("/tmp"), ARCHIVE_FILE_NAME);
+      final File dump = Files.createTempFile(ARCHIVE_FILE_NAME, ".tar.gz").toFile();
+      exportVersionFile(tempFolder);
+      dumpConfigs(tempFolder);
+      dumpDatabase(tempFolder);
+
+      Archives.createArchive(tempFolder, dump.toPath());
+      return dump;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public void deleteOrphanDirectories() {
```

Comment:
per the conversation we had, i think that the atomic replace in config persistence should handle this for us. ideally we shouldn't need to call this. the db should just handle it.